pip/pep425tags.py

```diff
@@ -189,6 +236,11 @@ def get_supported(versions=None, noarch=False):
             else:
                 # arch pattern didn't match (?!)
                 arches = [arch]
+        elif sys.platform == 'linux':
+            if is_manylinux1_compatible():
+                arches = [arch, arch.replace('linux', 'manylinux1')]
```

Comment:
I did the test, and the current ordering make `pip install` select the `linux` arch over the `manylinux1` arch when both wheels are available. Switching the ordering at that line yield the opposite behavior.

- pro: this is backward compatible with the master behavior
- con: this is less specific, one could expect pip to select the most platform specific wheel instead.

Note that `pip wheel` does not seem to be impacted by this change as it calls `python setup.py bdist_wheel` in a subprocess (I think) and therefore always name the generated file with the `linux` arch. I think this is good as to generate a valid manylinux1 wheel file extract care is required (e.g. using `auditwheel repair` to embed system shared libraries not provided a manylinux1 compatible system).

@dstufft any opinion on this ordering issue?